components/builder-web/app/oauth-providers/index.ts

```diff
@@ -43,6 +44,8 @@ export abstract class OAuthProvider {
 
   static fromConfig(type: string, clientID: string, authorizeUrl: string, redirectUrl: string, signupUrl: string, state: string): OAuthProvider {
     switch (type) {
+      case OAuthProviderType.ActiveDirectory:
+        return new ActiveDirectoryProvider(clientID, authorizeUrl, redirectUrl, signupUrl, state);
```

Comment:
Same comment as below here: I'd just ask that we be sure these arguments are limited to those required specifically by this provider.

Reply:
I tried limiting them (for example removing the signup url) and the typechecker went crazy. It looks like this is duplicated in every provider from what I can tell.